docs/guides.md

```diff
@@ -523,3 +523,89 @@ Now we enable autostart at the virtual machine level.
 5. Checking the output 
 `# xe vm-param-list uuid=<VM_UUID> | grep other-config`
 
+## Guest UEFI Secure Boot
+
+Enabling UEFI Secure Boot for guests ensures that XCP-ng will ensure that guest VMs only execute trusted binaries.  In practice, these are the binaries released by the operating system (OS) team for the OS in running in the VM (Microsoft Windows, Debian, Alpine, etc.).
+
+### Requirements
+
+* A UEFI guest VM.
+
+### Windows
+
+The steps in this guide are all that are necessary to enable secure boot.  Just follow the instructions provided here.
+
+### Linux
+
+It is required to follow the Linux distribution's instructions for enabling UEFI Secure Boot by installing shim and the distribution's signed binaries.  First follow the instructions from the distribution, then follow the instructions provided here.
+
+### Configure the Host
+
+Before enabling UEFI Secure Boot for guest VMs, first execute the `secureboot-certs` script.  This tool downloads, formats, and installs the publically available certificates from Microsoft.
+
+From the XCP-ng CLI:
+
+```
+# Download and install MS certificates
+secureboot-certs
+```
+
+:::tip
+Running `secureboot-certs` only needs to be done once per host, not per VM.
+:::
+
+### Configure the Guest VM
```

Comment:
We should differentiate between new VMs and existing VMs.

For new VMs, it is possible to enable secureboot directly from XO at VM creation (see advanced parameters). According to what I read at https://docs.citrix.com/en-us/citrix-hypervisor/vms/windows.html and https://docs.citrix.com/en-us/citrix-hypervisor/vms/troubleshoot.html#troubleshoot-uefi-and-secure-boot-problems-on-windows-vms I don't think `varstore-sb-state` is needed for new VMs provided you don't boot them in another mode before.

I also see at https://docs.citrix.com/en-us/citrix-hypervisor/vms/troubleshoot.html#troubleshoot-uefi-and-secure-boot-problems-on-windows-vms that VMs may fail to start if they have unsigned drivers. Luckily, I think our PV drivers are signed. I have you tested with them or only with Citrix drivers?

Reply:
Sure.  If the VM has not been booted (in UEFI mode), is not a vm-import, and the user has not called varstore-set, then varstore-sb-state setup is not required.

Regarding them failing to start with unsigned drivers, that is correct and is expected behavior.  I've only tested with our test VMs and from clean installs.  If we have our own drivers that are not signed by MS, we need to include our own certificate in `/var/uefistored/db.auth` as well.  I didn't even know Windows has a Xen PV mode, thanks for bringing that up.

Reply:
Note: it's HVM + PV drivers rather than pure PV.